crates/sui-rosetta/src/construction.rs

```diff
@@ -291,8 +309,9 @@ pub async fn metadata(
         return Err(Error::TransactionDryRunError(error.to_string()));
     }
 
-    let budget =
-        effects.gas_cost_summary().computation_cost + effects.gas_cost_summary().storage_cost;
+    let budget = budget.unwrap_or(
+        effects.gas_cost_summary().computation_cost + effects.gas_cost_summary().storage_cost,
+    );
```

Comment:
Since now we do dry run before the execution step, we could skip the dry run here if we don't need to calculate the budget